src/deploy/slash-indicator-notified-migrator.ts

```diff
@@ -0,0 +1,48 @@
+import { network } from 'hardhat';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+import { roninchainNetworks } from '../configs/config';
+import { DEFAULT_ADDRESS, Network } from '../utils';
+
+const deploy = async ({ getNamedAccounts, deployments, ethers }: HardhatRuntimeEnvironment) => {
+  if (!roninchainNetworks.includes(network.name!)) {
+    return;
+  }
+
+  // No need to deploy on hardhat network, to not break fixture
+  if (network.name === Network.Hardhat) {
+    return;
+  }
+
+  const { deploy } = deployments;
+  const { deployer } = await getNamedAccounts();
+
+  const proxyDepl = await deployments.get('SlashIndicatorProxy');
+  const newImplDelp = await deployments.get('SlashIndicatorLogic');
+
+  const IMPLEMENTATION_SLOT = '0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc';
+  const prevImplRawBytes32 = await ethers.provider.getStorageAt(proxyDepl.address, IMPLEMENTATION_SLOT);
+  const prevImplAddr = '0x' + prevImplRawBytes32.slice(26);
+  const roninValidatorAddr = await deployments.get('RoninValidatorSetProxy');
```

Comment:
bring it to a helper to reduce duplicated code

Reply:
updated.